### ISSUE TITLE
[feature]: Add Contact Form

### ISSUE THREAD

**Issue by sujoy-coder** [garg3133/JagratiWebApp#165]
### Is your feature request related to a problem?
I would like to work on the issue . To add a contact form with in home page for getting the user feedback it should be fully functioning  👀

**Comment by Naman-12911** [comment]
I want develop this feature under GSSOC please assign me

**Comment by aroma-sketch** [comment]
hey there, I am new to open source projects, and I would like to solve this issue as a small step, looking ahead to a response. it would be helpful .

**Comment by sujoy-coder** [comment]
Ok @vdmehta36, I am started working on it

**Comment by tmo66** [comment]
Hey @sujoy-coder, I'm glad that you choose our project. But we planned to shift to a new Home Page, so your hard work may go in vain. So I suggest you wait for some time until #160 gets merged. Then you can create a contact us section on that Home Page.

I will update you whenever it gets merged.

**Comment by tmo66** [comment]
And @Naman-12911 and @aroma-sketch as @sujoy-coder opened the issue and asked first he should get the first chance.
You can claim other unassigned issues, or can find bugs in the project!

**Comment by sujoy-coder** [comment]
Okay @tmo66 , no problem.
I will be waiting for your update ☺

**Comment by garg3133** [comment]
Hey @Naman-12911 @aroma-sketch, you can take a look at some other issues on this repository and if you find some issue you can work on and that issue has been inactive for more than 5 days (It is assigned to someone but no PR is created for that issue), you can get that issue assigned to you :)

And if you don't find any, you can let me know the domain on which you want to work and I'll create some new issues for you.

**Comment by aroma-sketch** [comment]
sure @garg3133, can u tell which one is inactive issue? I would like to work on this issue.

**Comment by garg3133** [comment]
@aroma-sketch You can take a part of the issue #124 as mentioned here: https://github.com/garg3133/JagratiWebApp/issues/124#issuecomment-793471639. Just quote and reply on my comment on that issue and I'll assign it to you. 

This is a very straightforward issue and will surely help you in getting started with Open Source.

**Comment by simranbhalla3** [comment]
Hii ,  I am a Gssoc participant and would like to work on this issue. could you Please assign this to me? @garg3133

**Comment by sujoy-coder** [comment]
Hey @tmo66 ,what is the progress of your PR on issue #160 ...
Please let me know otherwise I will be late to submit the PR for this issue 🙂

**Comment by garg3133** [comment]
Sorry @sujoy-coder, I almost forgot about the PR #167 for that issue. I've posted a review comment on it and it'll possibly be mered very soon :)

**Comment by sujoy-coder** [comment]
Okay @garg3133

**Comment by garg3133** [comment]
PR #167 has been merged now.

**Comment by sujoy-coder** [comment]
ok

**Comment by tmo66** [comment]
Hey @sujoy-coder, I've left the contact us section in my PR. 

This is how we are planning for the contact us part 
https://www.figma.com/proto/g83XJFmTssvU0qXBnlfIyO/Jagrati?node-id=0%3A3&scaling=scale-down-width, 
you are free to add your creative inputs, as long as they sync with the theme and the style of the home page.

Feel free to ping me wherever you need any help or want to discuss about the layout.

**Comment by sujoy-coder** [comment]
Yeah I have started working on it. Don't worry I will attach some screenshots how it is looking before making PR. ✌

**Comment by sujoy-coder** [comment]
Hey @tmo66 ,
##### Please check the design of the contact form.

### Here are some features which comes with this contact form
- [x] validating contact details before save in database
- [x] save the contact details in the database
- [x] any user can access this contact form (Not authenticated user as well)
- [x] write clean and formatted code

 

![Screenshot (108)](https://user-images.githubusercontent.com/65298629/111907658-53b6d180-8a7c-11eb-97b9-6d73ebc5c4c9.png)


Please let me know if any changes required 🙏

**Comment by tmo66** [comment]
@sujoy-coder, you don't have to make changes to this Home Page. Go to `127.0.0.1:8000/new`  and create a new contact us section in "new home page" similar to that of Figma File.

**Comment by sujoy-coder** [comment]
Oo I will change that. But the design is okay or I have to do the exact same design of Figma File ?

**Comment by sujoy-coder** [comment]
Here is the modified version

Please have a look...

![Screenshot (129)](https://user-images.githubusercontent.com/65298629/111953801-1ea68f80-8b0d-11eb-9656-eae04b24a4fd.png)

**Comment by tmo66** [comment]
Hey @sujoy-coder, I think it can be improved in many ways.

- You can remove the background of the Contact us Form, it would be much better that way.
- Then add some border-radius, shadow, and other CSS stuff for Text areas.
- Align the heading and the input area. Ex: Name is a little bit left of the text area of it.
- Sync Submit button with other buttons present on the homepage, it's basically a class so you can use that class.
- Interchange the position of the Form and the contact details. The form should be on the left side and vice versa.
- The icons you use for mail, the phone should be of the color matching with the button because that's the main theme of the home page.
- Align elements properly, as the information goes just below of location icon, while it should be aligned with Pandit. Also, give proper space between icons and the texts.
- I'm not sure about the Map, it's looking weird in that way. We have to add Social Media Icons too, so think about them too.

**Comment by sujoy-coder** [comment]
Now Check this one.

And let me know if any changes required...

![Screenshot (130)](https://user-images.githubusercontent.com/65298629/112110879-81fdf380-8bd9-11eb-8131-51ab8ac10c5e.png)

**Comment by tmo66** [comment]
Looking much better, isn't it? Still a few modifications

- I'm talking about the shadow in text areas, not in the whole division. 
- Ahh, you did a typo in the Address.

Could you add a Screenshot of how it looks like on a smaller screen?

P.S. Hide the number! It's a personal one...

**Comment by sujoy-coder** [comment]
Okay I am doing the modifications

**Comment by sujoy-coder** [comment]
### Here is two kind of shadow effect. Please let me know which one should I apply

## Type -1

![Screenshot (137)](https://user-images.githubusercontent.com/65298629/112121700-86c8a480-8be5-11eb-9537-d1cbc7fc0b01.png)

## Type-2

![Screenshot (142)](https://user-images.githubusercontent.com/65298629/112121882-bd062400-8be5-11eb-821d-60c2be3e3d66.png)

**Comment by tmo66** [comment]
Going with Type - 1! 
Just add border-radius, and increase the distance between form and contact details and Align `Jagrati` with contact details, and open a PR!!

@garg3133 Sir anything else required?

**Comment by sujoy-coder** [comment]
Okay 👍

**Comment by garg3133** [comment]
I liked the third design of the contact form section more (the one with the shadow in the whole form). It just required a bit of padding around the form and some border-radius.

For the second section, there should be some margin below Jagrati and "Our social media". Also, Our Social Media can be replaced with "Connect with us on".

Also, I think the contact details would look better when center-aligned instead of left-aligned.

Also, how about swapping the positions of both sections, making the boundary of contact details section just the form section (in the form of cards) and then overlapping the two cards at the ends (right end of contact details and left end of contact form). Wouldn't it look good?

**Comment by sujoy-coder** [comment]
@tmo66 What do you think?

**Comment by sujoy-coder** [comment]
Hey @garg3133 I am not able to understand this could you clarify :
`Also, how about swapping the positions of both sections, making the boundary of contact details section just the form section (in the form of cards) and then overlapping the two cards at the ends (right end of contact details and left end of contact form).`

Thank You...

**Comment by garg3133** [comment]
@sujoy-coder I mean something like this:
![contact-us](https://user-images.githubusercontent.com/39924567/112142018-16794d80-8bfc-11eb-99bf-612ee0819ea0.jpeg)

**Comment by sujoy-coder** [comment]
Okay @garg3133 I got you

**Comment by sujoy-coder** [comment]
@garg3133 is it okay ? or some modification required...
Please check it 

![Screenshot (143)](https://user-images.githubusercontent.com/65298629/112147205-6e1ab780-8c02-11eb-9681-bc355f5d8b36.png)

**Comment by garg3133** [comment]
I was talking about the above design of the form and not the latest one (the one with the orange border around the text inputs instead of shadows).

Also, increase the width of the cards, these look a bit elongated. And add some border-radius on both the cards (something like 5px).

Add some margin above Jagrati to bring it to the same level as Name label.

Other than that, it looks good to me.
